commit 30551fb3ab39134093ce94b777801487b26006fa (refs/changes/44/1344/3)
Author: Josh Wilsdon <jwilsdon@joyent.com>
Date:   2017-01-25T19:48:46-08:00 (2 years, 8 months ago)
    
    TOOLS-1662 sdcadm add-new-agent-svcs fails with: TypeError: Cannot read property 'params' of undefined

diff --git a/CHANGES.md b/CHANGES.md
index fc1a9fd..b4d5931 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -5,11 +5,16 @@
 -->
 
 <!--
-    Copyright 2016 Joyent, Inc.
+    Copyright 2017 Joyent, Inc.
 -->
 
 # sdcadm Changelog
 
+## 1.15.2
+
+- TOOLS-1662: Fix 'sdcadm experimental add-new-agent-svcs' when adding new
+  services.
+
 ## 1.15.1
 
 - TOOLS-1633: Fix 'sdcadm up' of the assets zone. It was broken in
diff --git a/lib/cli/do_add_new_agent_svcs.js b/lib/cli/do_add_new_agent_svcs.js
index 62e782c..dadffc2 100644
--- a/lib/cli/do_add_new_agent_svcs.js
+++ b/lib/cli/do_add_new_agent_svcs.js
@@ -146,8 +146,7 @@ function do_add_new_agent_svcs(subcmd, opts, args, cb) {
                         }
                         if (!svcs.length) {
                             newAgentServices.push(agent);
-                        }
-                        if (!svcs[0].params.image_uuid) {
+                        } else if (!svcs[0].params.image_uuid) {
                             agentServices[agent] = svcs[0];
                             updateAgentServices.push(agent);
                         }
diff --git a/package.json b/package.json
index fe744b9..4f49452 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "sdcadm",
   "description": "Administer a SmartDataCenter (SDC) standup",
-  "version": "1.15.1",
+  "version": "1.15.2",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
